src/JustSaying/Messaging/Middleware/MiddlewareBuilder.cs

```diff
@@ -28,10 +32,15 @@ public static class MiddlewareBuilder
         /// <param name="middleware">The instances of <see cref="MiddlewareBase{TContext, TOut}"/> to add to the
         /// returned composite <see cref="MiddlewareBase{TContext, TOut}"/>.</param>
         /// <returns>A composite <see cref="MiddlewareBase{TContext, TOut}"/>.</returns>
-        public static MiddlewareBase<TIn, TOut> WithAsync<TIn, TOut>(this MiddlewareBase<TIn, TOut> inner, params Func<MiddlewareBase<TIn, TOut>, MiddlewareBase<TIn, TOut>>[] middleware)
+        public static MiddlewareBase<TIn, TOut> WithAsync<TIn, TOut>(
+            this MiddlewareBase<TIn, TOut> inner,
+            params Func<MiddlewareBase<TIn, TOut>, MiddlewareBase<TIn, TOut>>[] middleware)
         {
-            var policy = inner;
+            if (inner == null) throw new ArgumentNullException(nameof(inner));
+            if (middleware == null) throw new ArgumentNullException(nameof(middleware));
+            if (middleware.Any(x => x == null)) throw new ArgumentException("All provided middlewares should be non-null");
```

Comment:
Should provide the argument name here in addition to the message.

Reply:
Done 👍